packages/openapi-generator/src/options/options-per-service.spec.ts

```diff
@@ -75,7 +80,139 @@ describe('getServiceOptions', () => {
+            }
+          `);
+  });
+
+  it('builds PerService config with options per service.', async () => {
+    const config: OptionsPerService = {
+      'path/service': {
+        directoryName: 'dirName',
+        packageName: '@scope/package-name',
+        serviceName: 'serviceName'
+      }
+    };
+
+    mock({
+      path: {
+        'myConfig.json': JSON.stringify(config)
+      }
     });
+
+    await expect(
+      getOptionsPerService(
+        [{ absolutePath: '/user/path/service', relativePath: 'path/service' }],
+        { optionsPerService: 'path/myConfig.json' } as ParsedGeneratorOptions
+      )
+    ).resolves.toMatchInlineSnapshot(`
+            Object {
+              "path/service": Object {
+                "directoryName": "dirName",
+                "packageName": "@scope/package-name",
+                "serviceName": "serviceName",
+              },
+            }
+          `);
+  });
+
+  it('builds PerService config with partial options  per service.', async () => {
+    const partialConfig = {
+      'path/service': {
+        directoryName: 'dirName',
+        packageName: '@scope/package-name'
+      }
+    };
+
+    mock({
+      path: {
+        'myPartialConfig.json': JSON.stringify(partialConfig)
+      }
+    });
+
+    await expect(
+      getOptionsPerService(
+        [{ absolutePath: '/user/path/service', relativePath: 'path/service' }],
+        {
+          optionsPerService: 'path/myPartialConfig.json'
+        } as ParsedGeneratorOptions
+      )
+    ).resolves.toMatchInlineSnapshot(`
+            Object {
+              "path/service": Object {
+                "directoryName": "dirName",
+                "packageName": "@scope/package-name",
+                "serviceName": "service",
+              },
+            }
+          `);
+  });
+
+  it('throws for conflicting service names if strict naming is on', async () => {
+    await expect(
+      getOptionsPerService(
+        [
+          {
+            absolutePath: '/user/path1/service',
+            relativePath: 'path1/service'
+          },
+          { absolutePath: '/user/path2/service', relativePath: 'path2/service' }
+        ],
+        {
+          strictNaming: true
+        } as ParsedGeneratorOptions
+      )
+    ).rejects.toMatchInlineSnapshot(
+      '[Error: The following service specs lead to non unique file names: path2/service. You can either introduce/adjust a operions-per-service.json or disable the strictNaming flag.]'
+    );
+  });
+
+  it('renames for conflicting service names if strict naming is off', async () => {
+    await expect(
+      getOptionsPerService(
+        [
+          {
+            absolutePath: '/user/path1/service',
+            relativePath: 'path1/service'
+          },
+          { absolutePath: '/user/path2/service', relativePath: 'path2/service' }
+        ],
+        {
+          strictNaming: false
+        } as ParsedGeneratorOptions
+      )
+    ).resolves.toMatchInlineSnapshot(`
```

Comment:
Consider using `.toEqual({...})` here, because this is easier to maintain. Also applies to the other tests.